pkg/kubelet/dockershim/helpers_linux_test.go

```diff
@@ -63,6 +67,45 @@ func TestGetSeccompSecurityOpts(t *testing.T) {
 	}
 }
 
+func TestGetSeccompSecurityOptsWithFeatureGate(t *testing.T) {
+	// Set SeccompRuntimeDefault feature gate
+	utilfeature.DefaultFeatureGate.Set("SeccompRuntimeDefault=true")
+	defer utilfeature.DefaultFeatureGate.Set("SeccompRuntimeDefault=false")
+	assert.True(t, utilfeature.DefaultFeatureGate.Enabled(features.SeccompRuntimeDefault),
+		"SeccompRuntimeDefault feature is expected to be enabled.")
+
+	tests := []struct {
+		msg            string
+		seccompProfile string
+		expectedOpts   []string
+	}{{
+		msg:            "No security annotations",
+		seccompProfile: "",
+		expectedOpts:   nil,
+	}, {
+		msg:            "Seccomp unconfined",
+		seccompProfile: "unconfined",
+		expectedOpts:   []string{"seccomp=unconfined"},
+	}, {
+		msg:            "Seccomp default",
+		seccompProfile: v1.SeccompProfileRuntimeDefault,
+		expectedOpts:   nil,
+	}, {
+		msg:            "Seccomp deprecated default",
+		seccompProfile: v1.DeprecatedSeccompProfileDockerDefault,
+		expectedOpts:   nil,
+	}}
+
+	for i, test := range tests {
+		opts, err := getSeccompSecurityOpts(test.seccompProfile, '=')
+		assert.NoError(t, err, "TestCase[%d]: %s", i, test.msg)
+		assert.Len(t, opts, len(test.expectedOpts), "TestCase[%d]: %s", i, test.msg)
+		for _, opt := range test.expectedOpts {
+			assert.Contains(t, opts, opt, "TestCase[%d]: %s", i, test.msg)
```

Comment:
Perhaps, it would be easier to use `reflect.DeepEqual()` instead?

Reply:
I just following the same pattern from TestGetSeccompSecurityOpts() above to be consistent.